core/domain/prod_validation_jobs_one_off.py

```diff
@@ -1373,6 +1373,70 @@ def _get_custom_validation_functions(cls):
             ]
 
 
+class SkillOpportunityModelValidator(BaseSummaryModelValidator):
+    """Class for validating SkillOpportunityModel."""
+
+    @classmethod
+    def _get_model_domain_object_instance(cls, item):
+        return (
+            opportunity_services.get_skill_opportunity_from_model(item))
+
+    @classmethod
+    def _get_external_id_relationships(cls, item):
+        return {
+            'skill_ids': (
+                skill_models.SkillModel, [item.id])
+        }
+
+    @classmethod
+    def _validate_question_count(cls, item):
+        """Validate that question_count matches the number of questions linked
+        to the opportunity's skill.
+
+        Args:
+            item: ndb.Model. SkillOpportunityModel to validate.
+        """
+        skill_model_class_model_id_model_tuples = (
+            cls.external_instance_details['skill_ids'])
+
+        for (_, _, skill_model) in (
+                skill_model_class_model_id_model_tuples):
+            if skill_model is None or skill_model.deleted:
+                continue
+            skill = skill_services.get_skill_from_model(
+                skill_model)
+            question_skill_links = (
+                question_services.get_question_skill_links_of_skill(
+                    skill.id, skill.description))
```

Comment:
Yeah, because in topic editor, when the questions are fetched, we need to show to which skill they  belong to, and hence the skill descriptions of the corresponding IDs are also needed.

Reply:
@aks681 I might be missing something here, too. Shouldn't the description be derivable from the ID?